lib/auth/helpers.go

```diff
@@ -779,19 +779,14 @@ func NewTestTLSServer(cfg TestTLSServerConfig) (*TestTLSServer, error) {
 	}
 	tlsConfig.Time = cfg.AuthServer.Clock().Now
 
-	accessPoint, err := NewAdminAuthServer(srv.AuthServer.AuthServer, srv.AuthServer.AuditLog)
-	if err != nil {
-		return nil, trace.Wrap(err)
-	}
-
 	srv.Listener, err = net.Listen("tcp", "127.0.0.1:0")
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 
 	srv.TLSServer, err = NewTLSServer(context.Background(), TLSServerConfig{
 		Listener:      srv.Listener,
-		AccessPoint:   accessPoint,
+		AccessPoint:   srv.AuthServer.AuthServer.Cache,
```

Comment:
Weirdly, `NewTestTLSServer` was using `ServerWithRoles` as `AccessPoint` for `NewTLSServer` - in the real server spin up, we use authServer.Cache for this. I'm assuming that providing ServerWithRoles within tests was effectively a bug. I could be wrong tho!

Reply:
I'd guess @espadolini or @fspmarshall might have insights into this.